### DISCUSSION TITLE
Flashed Wrong Firmware on BetaFPV Meteor 75 Pro

### DISCUSSION THREAD

**Discussion by robotgoat** [ExpressLRS/ExpressLRS#2742] [Q&A]
Hi, I recently purchased a BetaFPV Meteor Pro with the ELRS serial receiver AIO. The all in one board that the Meteor 75 Pro has is the Betafpv F4 5A AIO: https://betafpv.com/collections/brushless-flight-controller/products/f4-1s-5a-aio-brushless-flight-controller-elrs-2-4g. 

I mistakenly updated the RX via WiFi using the wrong firmware file. I used the DIY Devices 2.4G > Generic ESP8525 Full-duplex RX as TX firmware thinking that it's the correct one as that's what the closest name to the stock target is for the AIO board. After looking up the target for the board itself, I realized too late that I had flashed the wrong firmware. 

Nonetheless, with the incorrect firmware now on the AIO board, I cannot flash the correct firmware. No matter what I do, all attempts to flash through ELRS Configurator v1.6.1 fail. The board will also not go into WiFi mode after 60 seconds. After attempting to flash, and fail flashing through ELRS Configurator, the green RX LED starts slow blinking with no change as long as it's powered up. Power cycling the board will make the green slow flashing LED disappear until I try flashing again through ELRS Configurator. 

I am using Linux Mint 21.3 for all these operations. I replicated the process on a Windows 11 device to receive the same errors messages, so I do not believe it is a host system issue. 

Is there any way to recover this board? I would appreciate any help. Thank you.

**Comment by BonAppetitEU** [comment]
Такая же проблема,поэтому буду искать и ждать решение вместе с вами. Спасибо

**Comment by anti-vaxxer** [comment]
I'm sorry to hear that. I don't know how to repair that. maybe try [these](https://github.com/ExpressLRS/ExpressLRS/assets/161294608/e4034c8b-efe0-45e0-a314-7a3bf1ea77aa) instructions I found [here](https://graysonhobby.com/betafpv-f4-1s-5a-aio-brushless-flight-controller-elrs.html)

**Reply by robotgoat** [reply]
Thank you! While those directions did not solve the issue, it helped me troubleshoot in the right direction. I somehow managed to goose the AIO into Wifi mode. I have not checked for reproducibility, but the steps I did are:

1. Connect board to computer.
2. Open ELRS Configurator and select the correct board target this time
3. Tick only the "Erase before flash" box and click Flash.
4. The log output should detect the board and begin attempting to flash. It will connect and give the following error message as the flash fails:
```
code:255
stderr:
stdout: ** Searching flight controllers ** > FC found from '/dev/ttyACM0' Detected the following serial ports on this system: /dev/ttyACM0 ======== PASSTHROUGH INIT ======== Trying to initialize /dev/ttyACM0 @ 420000 No CLI available. Already in passthrough mode?, If this fails reboot FC and try again! ======== RESET TO BOOTLOADER ======== * Using full duplex (CRSF) Cannot detect RX target, blindly flashing! esptool.py v4.2.1 Serial port /dev/ttyACM0 WARNING: Pre-connection option "no_reset" was selected. Connection may fail if the chip is not in bootloader or flasher stub mode. Connecting......................................
```
5. After this, don't do anything! The green RX LED will turn on and after a minute it will begin flashing rapidly. At last it is in Wifi mode. 
6. Download the correct firmware and upload to the AIO from Wifi. 
7. Success, board has been recovered!

**Reply by Ston31** [reply]
Thanks man for sharing that. I get the saaaame problem. I flashed the same firmware as you did because I could not find the right one I got by elrs webpage. 

What version u flashed once u got receiver i wifi mode back?

**Reply by Harsha308i** [reply]
I have the same problem but my receiver is able to go into wifi. But when I'm flashing the correct firmware and trying to upload the .bin.gz file(which it gives after flashing with ELRS configurator), it prompts me to upload .bin file only. I tried multiple times and also removed cache in my browser. Nothing seems to work. Can someone help me out ?

**Reply by pkendall64** [reply]
Upload the .bin.gz file. Do NOT unzip it.

**Reply by Harsha308i** [reply]
I'm not able to. It only allows me to upload .bin file

**Reply by pkendall64** [reply]
What version is running currently on the RX? If you can't update via wifi then you need to do the recovery flash via betaflight or a UART adapter

**Reply by Harsha308i** [reply]
I updated to latest 3.5.1 firmware. Btw how do you do the recovery flash via betaflight or a UART adapter ?

**Reply by pkendall64** [reply]
https://www.expresslrs.org/quick-start/unbricking/

You would be best asking on discord in the #help-and-support channel so the folk there can get you sorted much more quickly

**Comment by BonAppetitEU** [comment]
У меня положительного результата нет. :(

**Comment by BonAppetitEU** [comment]
Попробыл через Linux под Steam Deck и всё получилось,спасибо большое 👍

**Reply by robotgoat** [reply]
Glad to hear you were able to recover your board! I must say it is quite funny you used your Steam Deck and had success using that as your host system.

**Comment by BartlomiejAero** [comment]
F4 1S 12A AIO Brushless Flight Controller V3 ELRS

I updated the RX via Wi-Fi using the wrong firmware file. I used the DIY Devices 2.4G / Generic ESP8525 Full-duplex RX as TX.

The result? 

No possibility of binding to the TX equipment and in the WiFi network the AIO appeared as ExpressLRS TX instead of ExpressLRS RX.

Solution to the problem if the Wi-Fi connection is working and the LED is blinking very quickly?

We run the ExpressLRS Configurator.
Releases: 3.5.3 (latest at the moment)
Device category: Firmware recommended by the AIO manufacturer, which in my case is BETAFPV 2.4 GHz
Device: Firmware recommended by the AIO manufacturer, which in my case is BETAFPV 2.4 GHz AIO RX
Flaashing Method: Wi-Fi default 10.0.0.1 (AIO RX shows up as ExpressLRS TX)
BUILD

!!! ATTENTION VERY IMPORTANT !!!

We obtain a file called "firmware.bin.gz". 

**We change the name from "firmware.bin.gz" to "firmware.bin"**

We connect to the Wi-Fi AIO ExpressLRS TX.
We open the 10.0.0.1 page
UPDATE tab
**We load the file with the changed name**.
We run it.

There you go.

Reset and we get the ExpressLRS RX Wi-Fi connection again, and ExpressLRS Configurator informs us about the latest software in AIO RX. Success :)

I repeated the action once again, but with the correct "firmware.bin.gz" file without changing the name.

I pondered over this for hours, searched and thought it was all over.

Ultimately, the trick of changing the file name to whatever the configurator wants will work.

 TX 3.5.3 equipment has been bound to RX 3.5.3. 

Success :)

Good luck to anyone who ends up here.
